core/src/main/java/org/elasticsearch/common/network/NetworkService.java

```diff
@@ -273,4 +270,19 @@ public InetAddress resolvePublishHostAddresses(String publishHosts[]) throws IOE
         }
         return InetAddress.getAllByName(host);
     }
+
+    /**
+     * Register custom name resolver a DiscoveryPlugin might provide
+     * @param discoveryPlugins Discovery plugins
+     */
+    public static List<CustomNameResolver> registerCustomNameResolvers(Settings settings, List<DiscoveryPlugin> discoveryPlugins) {
```

Comment:
This is an odd name (it doesnt actually register anything), and I don't see how it belongs in NetworkService (it has to do with how node construction creates the NetworkService). I think it should be in Node (and it can then be private).